### PR TITLE
File Size dependent execution / resources for Tuebingen Clusters

### PR DESCRIPTION
**Pull request by apeltzer**
This adds:

- profile binac for our resources here
- base level definitions for requesting resources
- memory requests on input bam size (need to test this further) 

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [x] Make sure your code lints (`nf-core lint .`).
 - [ ] `CHANGELOG.md` is updated

**Learn more about contributing:** https://github.com/nf-core/deepvariant/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

**Comment by apeltzer** [comment]
@PhilPalmer please have a look :-) We should test the file.size() dependent execution .-)

**Comment by PhilPalmer** [comment]
Thanks @apeltzer thats great. Theres just a couple of things I'd like to check so I'll make some comments on those

File: conf/base.config [hunk 1]
```diff
@@ -21,6 +21,13 @@ process {
   maxRetries = 1
   maxErrors = '-1'
 
+  withName:preprocess_bam {
```
Comment (PhilPalmer) [inline conf/base.config]: I think this should be for the `make_examples` process as that is the most computationally intensive process, however, its probably worth doing it for all of the main deepvariant processes

File: conf/base.config [hunk 2]
```diff
@@ -21,6 +21,13 @@ process {
...
+  cpus = { check_max( 20, 'cpus' ) }
+  memory = { check_max( bam.size() * 5.GB  * task.attempt, 'memory'}
```
Comment (PhilPalmer) [inline conf/base.config]: I think you're missing a closing bracket, so it should be:
`memory = { check_max( bam.size() * 5.GB  * task.attempt, 'memory')}`

File: conf/base.config (hunk 1 above)
Comment (apeltzer) [reply conf/base.config]: Yes, it is :-)

File: conf/base.config (hunk 2 above)
Comment (apeltzer) [reply conf/base.config]: Yup, sorry :-/

**Comment by apeltzer** [comment]
Please change whatever you need to change and let me know :-)

File: conf/base.config (hunk 2 above)
Comment (PhilPalmer) [reply conf/base.config]: No problem

**Comment by PhilPalmer** [comment]
Thanks, I made those changes so it should be good now.

Is deepvariant still running from the job you submitted earlier?

**Pull request merged**